pkg/restore/restorers/pv_restorer.go

```diff
@@ -56,66 +57,46 @@ func (sr *persistentVolumeRestorer) Prepare(obj runtime.Unstructured, restore *a
 	delete(spec, "claimRef")
 	delete(spec, "storageClassName")
 
-	pvName, err := collections.GetString(obj.UnstructuredContent(), "metadata.name")
-	if err != nil {
-		return nil, nil, err
+	if boolptr.IsSetToFalse(backup.Spec.SnapshotVolumes) {
+		// The backup had snapshots disabled, so we can return early
+		return obj, nil, nil
 	}
 
-	// if it's an unsupported volume type for snapshot restores, we're done
-	if sourceType, _ := kubeutil.GetPVSource(spec); sourceType == "" {
+	if boolptr.IsSetToFalse(restore.Spec.RestorePVs) {
+		// The restore has pv restores disabled, so we can return early
 		return obj, nil, nil
 	}
 
-	restoreFromSnapshot := false
-
-	if restore.Spec.RestorePVs != nil && *restore.Spec.RestorePVs {
-		// when RestorePVs = yes, it's an error if we don't have a snapshot service
-		if sr.snapshotService == nil {
-			return nil, nil, errors.New("PV restorer is not configured for PV snapshot restores")
-		}
+	// Past this point, we expect to be doing a restore
 
-		// if there are no snapshots in the backup, return without error
-		if backup.Status.VolumeBackups == nil {
-			return obj, nil, nil
-		}
-
-		// if there are snapshots, and this is a supported PV type, but there's no
-		// snapshot for this PV, it's an error
-		if backup.Status.VolumeBackups[pvName] == nil {
-			return nil, nil, errors.Errorf("no snapshot found to restore volume %s from", pvName)
-		}
-
-		restoreFromSnapshot = true
+	if sr.snapshotService == nil {
```

Comment:
so this is going to return an error if SnapshotVolumes is auto for both the backup and the restore, and there was no snapshotService for the backup (so no backups).  I think this is definitely not desired behavior. I think we want to exit prior to this (without error) if `backup.status.VolumeBackups` is nil/empty, right? (we may have discussed this already this AM).

Reply:
Blech. That wasn't a case I was thinking about. I'll add that check.